File: SooYeon/231218_week/Programmers_77484_로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    var answer = [];
    
    let match_idx = []; // idx
    let zero_cnt = 0, match_cnt = 0;
    
    for(let i = 0; i < lottos.length; i++) {
        if(lottos[i] === 0) {
            zero_cnt++;
            continue;
        }
        
        let idx = win_nums.indexOf(lottos[i]);
        if(idx >= 0) {
            match_cnt++;
            match_idx.push(idx);
        }
    }
    

    // 최고 순위는 0들을 모두 당첨 번호로 바꾸는 것 -> 7 - match_idx + zero_cnt
    let high = 7 - (match_cnt + zero_cnt);
    if(high > 6) high = 6;
    answer.push(high);
        
    // 최저 순위는 0들이 모두 안맞는 것 ->  7 - match_idx
    let low = 7 - match_idx.length;
    if(low > 6) low = 6;
    answer.push(low);

    
    return answer;
}

File: MinJu/240108_week/Programmers_77484_로또의 최고 순위와 최저 순위.py
# 로또의 최고 순위와 최저 순위

# 6개 번호 모두 맞으면 1등 65432 2개 맞으면 5등 그 이외는 낮첨
# 알아볼 수 없는 번호 0 -> 최고, 최저 순위
def solution(lottos, win_nums):
    answer = []
    rank = [6, 6, 5, 4, 3, 2, 1] # 가능한 순위
    num = lottos.count(0)
    cnt = 0
    for i in win_nums:
        if i in lottos:
            cnt += 1
    # 최고 순위는 모르는 숫자도 모두 맞춘 것
    answer.append(rank[num+cnt])
    # 최저 순위는 모르는 숫자는 모두 못찾운 것
    answer.append(rank[cnt])
    return answer